### PR TITLE
updated link to jdk download archive

### LINKED ISSUE #44: Update link to jdk download source
I just noticed that the archive page cited in the build reference (https://adoptopenjdk.net/) has moved to a different page:

screenshot

The new link is https://adoptium.net/temurin/releases

### PR DESCRIPTION
**Pull request by jo-mueller**
Fixes #44 

Just a minor change to update a link to download JDK16

### CODE REVIEW

**Comment by petebankhead** [comment]
Thanks, I'm a bit undecided on this: I was planning to update the build instructions for v0.4.0 around the time of its release. It will likely use JDK 17, so the link should definitely be updated for that - but more build changes might also be necessary.

Meanwhile, the version tagged v0.3.x *won't* build using JDK 17 (because gradlew is rather strict about that kind of thing), and also might not work with earlier JDKs because of jpackage differences. And some people checking the build instructions could be looking to build the last released version rather than the current main branch.

So the fact that the current link is slightly out of date - but still leads to Java 16 rather than 17 by default - is arguably a useful feature not a bug... at least until the next QuPath release is closer to being ready and the build instructions revised more fully (hopefully in a month or two).

Aside from being slightly out of date (but still functional), does the current link cause any problems for anyone trying to follow the build instructions, which updating it resolves?

**Comment by jo-mueller** [comment]
In my case I used the link to build QuPath from the current main branch with JDK 16 - By replacing the link I only meant to cut getting to the download page short by one crosslink. But you are right, the link in the PR defaults to JDK 17 which could lead to trouble. 

I could change it to https://adoptium.net/temurin/releases/?version=16 - this would point to the current download source and so set the version to JDK 16?

**Comment by petebankhead** [comment]
It should now give the 'correct' link at https://qupath.readthedocs.io/en/latest/docs/reference/building.html

This won't be the 'stable' version that appears by default until QuPath v0.4.0 is released, but hopefully that won't be too far away.

I'll close this PR because I hope that works as the best long term solution. If I'd merged this, the changes would still have only appeared as 'latest' and not 'stable' because of how [readthedocs handles versioned documentation](https://docs.readthedocs.io/en/stable/versions.html). And then we have the confusion of Java 17 not being able to build QuPath v0.3. And if we're concerned about older versions and links then we should also update the one for v0.2 (which required Java 14)... so we'd have to have multiple PRs to different branches.

Basically, the various branches/versions on readthedocs become a bit confusing to maintain, so I'd rather just leave v0.3 (and v0.2) as they are - even with that extra cross-link - and go all-in towards getting the next release ready as soon as possible.

Thanks though for flagging this!
